rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,133 @@
+
+An analysis pipeline event handler component will be created that is responsible for partitioning a *data group* into co-processing units, dispatching the pipelines, and tracking completion. This is the responsibility of Analysis.  When analysis results are submitted to ingest, this creates a secondary analysis *data group*.
+
+The creation and update of *PROJECT_SUBMISSIONS* *data groups*, both primary and secondary, is the responsibility of the Ingest system.
+
+![Figure 2](../images/0000-multi-data-collection-data-processing-images/flow-of-information.png)
+Figure 2: Flow of information in the proposed data model
+
+#### Updates to *data groups*
+Updates to *PROJECT_SUBMISSIONS* must result in updating of the associated *data groups* to trigger reprocessing. When part or the entirety of a project submission is updated the associated data group must be identified and updated with the new bundle versions if bundles have been replaced and/or with additional new bundles if bundles have been added. This is the responsibility of Ingest.  Analysis must handle the update to the *data group* by initiating only the pipelines that have been affected by the update and ensure that the output bundles of this analysis are updates to existing bundles.
+
+*Data group* updates may result from creating new bundles to add additional types of data to a project, or updating bundles to change metadata or append data to data files (e.g. FASTQ top-up).
+
+#### Implementation feasibility
+
+Both Analysis and Ingest confirm that implementation of the data_groups bundle proposed here is feasible in principle. Analysis indicates that significant changes need to be made to the existing infrastructure to handle the new type of notification. The generation of secondary data_groups of type PROJECT_SUBMISSION will be generated by analysis submitting to submission and Ingest will identify the bundles included in the secondary bundle.
+
+The following order of implementation is proposed, accounting for the dependencies between components to ensure uninterupted system functionality:
+
+1) Ingest adds functionality for supporting data_groups and starts using them for singalling submission completion
+2) Analysis adds handler for data_groups but continues to use bundle events during a transition period
+3) Analysis switches to using data_group events
+4) Ingests adds support for secondary data_groups
```

Comment:
I think we can get up to here currently but going beyond requires resolution of the analysis updates RFC.